pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -626,7 +628,7 @@ public void start() throws PulsarServerException {
             this.bkClientFactory = newBookKeeperClientFactory();
 
             managedLedgerClientFactory = ManagedLedgerStorage.create(
-                config, localMetadataStore, getZkClient(), bkClientFactory, ioEventLoopGroup
+                config, localMetadataStore, createBookieZkClient(), bkClientFactory, ioEventLoopGroup
```

Comment:
Sorry, one last question.
who is going to `close` this ZooKeeper handle ?

when we use `getZkClient()` the ZooKeeper handle will be closed within the PulsarService lifecycle.
but now we are going to create instances of it without never shutting them down.

probably we have to store this reference to a field in PulsarService and dispose it in the "close()" method